test/e2e/e2ethanos/services.go

```diff
@@ -813,56 +702,56 @@ http {
 }
 `, tenantID, target)
 
-	dir := filepath.Join(e.SharedDir(), "data", "nginx", name)
-	if err := os.MkdirAll(dir, 0750); err != nil {
-		return nil, errors.Wrap(err, "create store dir")
+	f := e2e.NewInstrumentedRunnable(e, fmt.Sprintf("nginx-%s", name)).
+		WithPorts(map[string]int{"http": 80}, "http").
+		Future()
+
+	if err := os.MkdirAll(f.Dir(), 0750); err != nil {
+		return e2e.NewErrInstrumentedRunnable(name, errors.Wrap(err, "create store dir"))
 	}
 
-	if err := ioutil.WriteFile(filepath.Join(dir, "nginx.conf"), []byte(conf), 0600); err != nil {
-		return nil, errors.Wrap(err, "creating nginx config file failed")
+	if err := ioutil.WriteFile(filepath.Join(f.Dir(), "nginx.conf"), []byte(conf), 0600); err != nil {
+		return e2e.NewErrInstrumentedRunnable(name, errors.Wrap(err, "creating nginx config file failed"))
 	}
 
-	nginx := e2e.NewInstrumentedRunnable(e, fmt.Sprintf("nginx-%s", name)).WithPorts(map[string]int{"http": 80}, "http").Init(
+	return e2e.NewInstrumentedRunnable(e, fmt.Sprintf("nginx-%s", name)).WithPorts(map[string]int{"http": 80}, "http").Init(
```

Comment:
I think this is causing the CI to fail, I guess it should be `f.Init(...)` instead:
```suggestion
	return f.Init(
```

Reply:
DID you just save my day? ; p